File: packages/transform/src/features/protobuf/ProtobufAssertDecodeTransformer.ts
import { ProtobufAssertDecodeProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufAssertDecodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "protobuf.assertDecode",
      write: ProtobufAssertDecodeProgrammer.write,
    });
}

File: packages/transform/src/features/protobuf/ProtobufAssertDecodeTransformer.go
package protobuf

import (
	coreprotobuf "github.com/samchon/typia/packages/core/src/programmers/protobuf"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var ProtobufAssertDecodeTransformer = protobufAssertDecodeTransformerNamespace{}

type protobufAssertDecodeTransformerNamespace struct{}

func (protobufAssertDecodeTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coreprotobuf.ProtobufAssertDecodeProgrammer.Write(call.TypeText)
}
